From 03ca388ec3664af293486e6e61dbf4e06017c7aa Mon Sep 17 00:00:00 2001
From: Xiaofei Ma <xiaofei@motorola.com>
Date: Wed, 9 Sep 2015 15:10:53 -0500
Subject: [PATCH 142/959] IKQCOMSD1-2939: arm/dts: 8952: Toggle
 earpiece/speaker

One GPIO is used to switch earpiece and
speaker mode.

Change-Id: I72c05ab78e3ead2ca32bae7b05bce12ff511b627
Signed-off-by: Xiaofei Ma <xiaofei@motorola.com>
Reviewed-on: http://gerrit.mot.com/782313
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
index fea219d4043..d4f7e626ae0 100644
--- a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
@@ -74,6 +74,17 @@
 		};
 	};
 
+	tfa9890_earpiece_en {
+		qcom,pins = <&gp 68 0>;
+		qcom,num-grp-pins = <1>;
+		qcom,pin-func = <0>;
+		label = "tfa9890_earpiece";
+		tfa9890_earpiece_default: default {
+			bias-disable;
+			drive-strength = <2>;
+		};
+	};
+
 	spi4_active {
 		/* BLSP5 MOSI, MISO, CLK */
 		qcom,pins = <&gp 16>, <&gp 17>, <&gp 19>;
@@ -532,12 +543,13 @@
 	tfa9890@34 {
 		compatible = "nxp,tfa9890";
 		reg = <0x34>;
-		gpios = <&msm_gpio 94 0>;     /* RESET */
+		gpios = <&msm_gpio 94 0>,     /* RESET */
+			<&msm_gpio 68 0>;     /* Earpiece Enable */
 		tfa_vdd-supply = <&pm8950_l6>;
 		nxp,tfa_max-vol-steps = <15>;
 		nxp,tfa-dev = "left";
 		nxp,tfa_pcm-start-delay = <100>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&tfa9890_reset_default>;
+		pinctrl-0 = <&tfa9890_reset_default &tfa9890_earpiece_default>;
 	};
 };
-- 
2.11.0

